handsontable/src/plugins/formulas/engine/register.js

```diff
@@ -84,6 +86,13 @@ export function setupEngine(hotInstance) {
       });
     }
 
+    if (engineConfigItem.getConfig().leapYear1900 !== DEFAULT_SETTINGS.leapYear1900
+      || isObjectEqual(engineConfigItem.getConfig().nullDate, DEFAULT_SETTINGS.nullDate) === false) {
+      warn(toSingleLine`Dates stored in HyperFormula engine may don't cooperate properly with Handsontable when HF's 
+        configuration options such as \`leapYear1900\` and \`nullDate\` have been changed. Please consider using 
+        default configuration values for those keys.`);
```

Comment:
```suggestion
      warn(toSingleLine`If you use HyperFormula with Handsontable, keep the default \`leapYear1900\` and \`nullDate\` settings. Otherwise, HyperFormula's dates may not sync correctly with Handsontable's dates.`);
```